crates/api_crud/src/comment/update.rs

```diff
@@ -65,13 +65,19 @@ impl PerformCrud for EditComment {
     }
 
     if data.distinguished.is_some() {
-      // Verify that only a mod or admin can distinguish a comment
-      is_mod_or_admin(
```

Comment:
Why is the check failing tho? The admin field should not get federated anywhere, and they also aren't a mod of this community.

Reply:
Check the linked comment for an example. The existing code only checks if the user is mod/admin according to the mander.xyz database. So it happily lets the admin of that instance distinguish a comment in a remote community. This cant be allowed because we generally cant trust federated users (anyone could setup an instance and be admin).

It would be possible to only add the check in federation, and still let admins distinguish comments locally. But that would lead to bad UX imo because people expect that content is consistent across instances.

Reply:
Okay, I checked the DB, and luckily that user isn't an admin. So basically our apub **receiver** for distinguish comments is anyone remotely distinguish them, even if they aren't a local mod or admin. So the check should take place at that receiver.

Basically probably all apub receive actions that are doable by `admin_or_mod`, should only be doable by community mods. This is a security issue, because the apub receive could allow anyone to impersonate themselves as an admin and do an action.